references/scripts/create-reference-json.R

```diff
@@ -59,33 +62,74 @@ create_ref_entry <- function(
     mito_file = file.path(
       annotation_dir,
       glue::glue("{reference_name}.mitogenes.txt")
-    ),
-    t2g_3col_path = file.path(
-      annotation_dir,
-      glue::glue("{reference_name}.spliced_intron.tx2gene_3col.tsv")
-    ),
-    t2g_bulk_path = file.path(
-      annotation_dir,
-      glue::glue("{reference_name}.spliced_cdna.tx2gene.tsv")
-    ),
-    splici_index = file.path(
-      ref_dir, "salmon_index",
-      glue::glue("{reference_name}.spliced_intron.txome")
-    ),
-    salmon_bulk_index = file.path(
-      ref_dir, "salmon_index",
-      glue::glue("{reference_name}.spliced_cdna.txome")
-    ),
-    cellranger_index = file.path(
-      ref_dir, "cellranger_index",
-      glue::glue("{reference_name}_cellranger_full")
-    ),
-    star_index = file.path(
-      ref_dir, "star_index",
-      glue::glue("{reference_name}.star_idx")
     )
   )
 
+  # fill in values related to salmon/alevin-fry index
+  if (include_salmon) {
+    json_entry <- c(
+      json_entry,
+      list(
+        t2g_3col_path = file.path(
+          annotation_dir,
+          glue::glue("{reference_name}.spliced_intron.tx2gene_3col.tsv")
+        ),
+        t2g_bulk_path = file.path(
+          annotation_dir,
+          glue::glue("{reference_name}.spliced_cdna.tx2gene.tsv")
+        ),
+        splici_index = file.path(
+          ref_dir, "salmon_index",
+          glue::glue("{reference_name}.spliced_intron.txome")
+        ),
+        salmon_bulk_index = file.path(
+          ref_dir, "salmon_index",
+          glue::glue("{reference_name}.spliced_cdna.txome")
+        )
+      )
+    )
+  } else {
+    json_entry <- c(
+      json_entry,
+      list(
+        t2g_3col_path = NA,
+        t2g_bulk_path = NA,
+        splici_index = NA,
+        salmon_bulk_index = NA
+      )
+    )
+  }
```

Comment:
For a bit more clarity, and fewer `else` statements. I think I might modify the initial definitions to include all optional reference files set as `NA` to start, and then replace them in these blocks as needed:

```suggestion
  if (include_salmon) {
    json_entry$t2g_3col_path = file.path(
      annotation_dir,
      glue::glue("{reference_name}.spliced_intron.tx2gene_3col.tsv")
    )
    json_entry$t2g_bulk_path = file.path(
      annotation_dir,
      glue::glue("{reference_name}.spliced_cdna.tx2gene.tsv")
    )
    json_entry$splici_index = file.path(
      ref_dir, "salmon_index",
      glue::glue("{reference_name}.spliced_intron.txome")
    )
    json_entry$salmon_bulk_index = file.path(
      ref_dir, "salmon_index",
      glue::glue("{reference_name}.spliced_cdna.txome")
    )
  }
```